src/lib/__tests__/routing-test.ts

```diff
@@ -0,0 +1,152 @@
+import { createSearchClient } from '../../../test/mock/createSearchClient';
+import { wait } from '../../../test/utils/wait';
+import historyRouter from '../routers/history';
+import instantsearch from '../..';
+import { connectSearchBox } from '../../connectors';
+
+describe('routing', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('writeOnDispose=true', () => {
+    test('cleans URL on dispose', async () => {
+      const pushState = jest.spyOn(window.history, 'pushState');
+      const writeDelay = 400;
+
+      const search = instantsearch({
+        indexName: 'indexName',
+        searchClient: createSearchClient(),
+        routing: {
+          router: historyRouter({
+            writeDelay,
+          }),
+        },
+      });
+
+      search.addWidgets([connectSearchBox(() => {})({})]);
+
+      search.start();
+
+      // Wait for ${writeDelay} to pass then check URL has been initialized
```

Comment:
I'm not sure all the comments clarify anything, to me they're describing what the code already says.

Why do you think we need cooments here?

Reply:
Sometimes we need to check if the URL was initialized/update/cleaned (after a dispose), that's why this comment is for, but I guess I can remove the beginning "Wait for ${writeDelay} to pass then" which is common for every sentence